docs/technical-overview/caching.md

```diff
@@ -0,0 +1,70 @@
+---
+sidebar_position: 8
+---
+
+# Caching
+
+_This has been adapted from [The Three Caches of Forem // Take on Rules](https://takeonrules.com/2022/03/29/the-three-caches-of-forem/)_
+
+
+In the Forem code base, we make extensive use of various caching strategies. And as with any cache, we always run the risk of not invalidating the right caches.
+
+The three caching strategies are:
+
+- Edge Caching
+- Rails Caching
+- Database Caching
+
+At a high level, we leverage caches to improve performance. Let’s delve into each strategy. But before we do, let’s introduce a few concepts.
+
+## Response Documents and Fragments
+
+The <dfn>response document</dfn> is a single file sent from the server in response to a request.
+
+Remember that what you see rendered in your browser is almost always from the combination of many _response documents_ sent from the server: <abbr title="Hypertext Markup Language">HTML</abbr> documents, Javascript files, <abbr title="Cascading Stylesheet">CSS</abbr> files, and <abbr title="XMLHttpRequest">XHR</abbr> that modify the <abbr title="Document Object Model">DOM</abbr>.
+
+To build a single _response document_, the server often assembles multiple fragments to form that singular document. In Ruby on Rails this is done with like layouts, views, and partial views.
+
+## Back to Listing the Caching Strategies
+
+Let’s now work from the broader caching strategies to the more narrow ones.
+
+### Edge Caching
+
+Edge caching is about trying to put as many of the _response documents_ as close to the client as possible to reduce latency on content delivery.
+
+At Forem, we enable the usage of either Fastly or Nginx. When we bust the edge cache, we are evicting specific _response documents_.
+
+The next time someone requests that _response document_ the server will reassemble it (from the various constituent parts).
+
+You can see [our Edge Busting strategy here](https://github.com/forem/forem/blob/3ddda918cb40f8bfb7fa8ad7dae5736717872bc8/app/services/edge_cache/bust.rb).
```

Comment:
```suggestion
You can see [our Edge Busting strategy here](https://github.com/forem/forem/blob/main/app/services/edge_cache/bust.rb).
```

Should we point this at whatever the current version of the file on `main` is?

Reply:
My preference is to point to the SHA as things change.  But I can see that "as things change we should update docs"